src/rust/bin/wallet_tool.rs

```diff
@@ -0,0 +1,528 @@
+            ));
+            e
+        })
+}
+
+fn prompt<'a>(input: &mut Stdin, buf: &'a mut String) -> io::Result<&'a str> {
+    let n = input.read_line(buf)?;
+    if n == 0 {
+        return Err(io::Error::from(ErrorKind::UnexpectedEof));
+    }
+    Ok(buf.trim_end_matches(|c| c == '\r' || c == '\n').trim())
+}
+
+fn ordinal(num: usize) -> String {
+    let suffix = if (11..=13).contains(&(num % 100)) {
+        "th"
+    } else {
+        match num % 10 {
+            1 => "st",
+            2 => "nd",
+            3 => "rd",
+            _ => "th",
+        }
+    };
+    format!("{}{}", num, suffix)
+}
+
+fn zcash_cli_path(debug: bool) -> io::Result<PathBuf> {
+    // First look for `zcash_cli[.exe]` as a sibling of the executable.
+    let mut exe = env::current_exe()?;
+    exe.set_file_name("zcash-cli");
+    exe.set_extension(EXE_EXTENSION);
+    if debug {
+        eprintln!("DEBUG: Testing for zcash-cli at {:?}", exe);
+    }
+    if exe.exists() {
+        return Ok(exe);
+    }
+    // If not found there, look in `../src/zcash_cli[.exe]` provided
+    // that `src` is a sibling of `target`.
+    exe.pop(); // strip filename
+    exe.pop(); // ..
+    if exe.file_name() != Some(OsStr::new("target")) {
+        // or in `../../src/zcash_cli[.exe]` under the same proviso
+        exe.pop(); // ../..
+        if exe.file_name() != Some(OsStr::new("target")) {
+            return Err(io::Error::from(ErrorKind::NotFound));
+        }
+    }
+    // Replace 'target/' with 'src/'.
+    exe.set_file_name("src");
+    exe.push("zcash-cli");
+    exe.set_extension(EXE_EXTENSION);
+    if debug {
+        eprintln!("DEBUG: Testing for zcash-cli at {:?}", exe);
+    }
+    if exe.exists() {
+        Ok(exe)
+    } else {
+        Err(io::Error::from(ErrorKind::NotFound))
+    }
+}
+
+fn exec(exe_path: &Path, args: &[String], debug: bool) -> Result<Output, io::Error> {
+    if debug {
+        eprintln!("DEBUG: Running {:?} {:?}", exe_path, args);
+    }
+    Command::new(exe_path).args(args).output()
+}
+
+fn default_filename_base() -> String {
+    let format = format_description!("export[year][month][day]");
+
+    // We use the UTC date because there is a security issue in obtaining the local date
+    // from either `chrono` or `time`: <https://github.com/chronotope/chrono/issues/602>.
+    // We could use the approach in
+    // <https://github.com/ArekPiekarz/rusty-tax-break/commit/3aac8f0c26fd96b7365619509a544f78b59627fe>
+    // if it were important, but it isn't worth the dependency on `tz-rs`.
+
+    OffsetDateTime::from(SystemTime::now())
+        .format(&format)
+        .unwrap_or_else(|_| "export".to_string())
+}
+
+fn clear_and_show_cautions(export_path: &str) {
+    try_to_clear(concat!(
+        "\nCAUTION: This terminal window might be showing secrets (or have\n",
+        "them in the scrollback). Please copy any useful information and\n",
+        "then clear it"
```

Comment:
```suggestion
        "then close it"
```
`clear` is not reliable and may retain scrollback. `reset` is better, but really I think the best suggestion is just to close the terminal session.

Reply:
As far as I know, `clear` does not retain scrollback on any terminals that support scrollback clearing (MATE terminal, `xterm`, and Linux VTs do at least). I agree that it isn't absolutely reliable: terminals are stateful and it is possible to be in a state where `clear` doesn't work, although then you wouldn't necessarily see the message either.

I didn't say "close" because it may be a VT that can't be closed, not a windowed terminal. Note that " using 'clear'." will be printed after this message (on non-Windows platforms). How about "then close it, or clear it using 'clear'."?

Reply:
I realized why I consider `clear` to not clear terminals; it's that I use tmux, and neither tmux nor GNU screen clears a terminal's scrollback buffer on either `clear` or (as I just discovered experimentally) `reset`. So it's specific to my setup, but given that using terminal multiplexers is relatively common, closing the terminal is probably safest.

Reply:
Clear on my Mac at least, only clears the screen. I can still scroll back to see prior Terminal activity.

Reply:
@steven-ecc Does the tool actually clear scrollback when run on mac OS? The message is only a fallback in case the attempt to clear using the `clearscreen` crate fails, so I think we should not worry about this if it works in practice.

Reply:
@steven-ecc Can you test whether pressing Command + K clears scrollback in a Mac terminal, please?